src/main/java/Duke.java

```diff
@@ -1,10 +1,206 @@
+
+    static void list() {
+        try {
+            listEmpty(taskList);
+            UI.printOutput((taskList));
+        } catch (ListEmptyException e) {
+            UI.printListEmpty();
+        }
+    }
+
+    static void done(String line) {
+        try {
+            listEmpty(taskList);
+            String theStr = line.substring(5);
+            String[] strArr = theStr.split(",");
+            int[] intArr = new int[strArr.length];
+            for (int i = 0; i < strArr.length; i++) {
+                String num = strArr[i];
+                intArr[i] = Integer.parseInt(num);
+                indexOutOfRange(taskList.size(), intArr[i]);
+            }
+
+            for (int i = 0; i < intArr.length; i++) {
+                Task t = taskList.get(intArr[i] - 1);
+                t.markAsDone();
+                UI.printMarkedAsDone(t);
+            }
+        } catch (NumberFormatException e) {
+            UI.printNumberFormatException();
+        }
+        catch (IndexOutOfRangeException e) {
+            UI.printIndexOutOfRangeException();
+        }
+        catch  (ListEmptyException e) {
+            UI.printListEmpty();
+        }
+    }
+
+    static void invalidTask(String line) {
+        try {
+            checkEmpty(line);
+            checkInvalidWord(line);
+        } catch (EmptyException e) {
+            UI.printEmptyException();
+        } catch (StringFormatException e){
+            UI.printStringFormatException();
+        }
+    }
+
+    static void checkEmpty(String description) throws EmptyException {
+        if (description.isEmpty()){
+            throw new EmptyException ();
+        }
+    }
+
+    static void checkInvalidWord(String description) throws StringFormatException {
+        if ( !( description.equals("bye") || description.equals("list")
+                || description.equals("todo") || description.equals("event")
+                || description.equals("deadline"))) {
+            throw new StringFormatException ();
+        }
+    }
+
+    static void checkEmptyToDoDescription (String description) throws EmptyToDoDescriptionException {
+        if (description.equals("todo")) {
+            throw new EmptyToDoDescriptionException();
+        }
+    }
+
+    static void checkEmptyEventDescription (String description) throws EmptyEventDescriptionException {
+        if (description.equals("event")) {
+            throw new EmptyEventDescriptionException();
+        }
+    }
+
+    static void checkEmptyDeadlineDescription (String description) throws EmptyDeadlineDescriptionException {
+        if (description.equals("deadline")) {
+            throw new EmptyDeadlineDescriptionException();
+        }
+    }
+
+    static void indexOutOfRange(int size,  int number) throws IndexOutOfRangeException {
+        if (number > size || number < 0) {
+            throw new IndexOutOfRangeException();
+        }
+    }
+
+    static void listEmpty( ArrayList<Task> taskList) throws ListEmptyException {
```

Comment:
Similarly, can consider verb or boolean forms

Reply:
Thanks for the advice, will improvise the code accordingly!